### PR TITLE
[python legacy] Stripping quotes in convert_string_to_expr() when predicate value is a string type

### LINKED ISSUE #3874: [BUG] convert_string_to_expr() from python_legacy module did not remove the quotes when the value is a string
```
>>> Expressions.equal('dt', '2021-12-21')
Predicate(Operation.EQ,ref(name="dt"),"2021-12-21")
>>> Expressions.convert_string_to_expr("dt='2021-12-21'")
Predicate(Operation.EQ,ref(name="dt"),"'2021-12-21'")
```
cc: @jun-he @TGooch44

### PR DESCRIPTION
**Pull request by puchengy**
This fixes #3874

### CODE REVIEW

**Comment by puchengy** [comment]
@jun-he @TGooch44 mind  taking a look?

**Review by wizardxz — APPROVED** [review]

**Review by wizardxz — APPROVED** [review]

File: python_legacy/iceberg/api/expressions/expression_parser.py [hunk 1]
```diff
@@ -143,8 +143,10 @@ def get_expr(node, expr_map):
         return mapped_op(*get_expr(node[op], expr_map))
     elif isinstance(node, (list, tuple)):
         return (get_expr(item, expr_map) for item in node)
-    elif isinstance(node, (str, int, float)):
+    elif isinstance(node, (int, float)):
         return node
+    elif isinstance(node, str):
+        return node.strip("'").strip('"')
```
Comment (puchengy) [inline python_legacy/iceberg/api/expressions/expression_parser.py:149]: I might need to double check on cases such as `"'abc"`

File: python_legacy/iceberg/api/expressions/expression_parser.py [hunk 2]
```diff
@@ -126,6 +126,15 @@ def get_expr_tree(tokens):
                  binary_tuples]}
 
 
+def handle_str_expr(node):
+    if node.startswith("'") and node.endswith("'"):
```
Comment (jun-he) [inline python_legacy/iceberg/api/expressions/expression_parser.py:130]: Will it handle some complicated cases, e.g. `'foo\'bar\'' case as well?
wondering if we can use some functions in pyparsing quotedString to solve it?

**Comment by rdblue** [comment]
@TGooch44 could you take a look at this since you're familiar with the parser?

**Comment by puchengy** [comment]
@rdblue @jun-he @TGooch44 change updated, mind taking another look? thanks.

**Review by rdblue — APPROVED** [review]

**Comment by rdblue** [comment]
Nice work. Thanks, @puchengy!

**Comment by jun-he** [comment]
Thanks @puchengy !

**Pull request merged**